components/dashboard/src/admin/BlockedRepositories.tsx

```diff
@@ -0,0 +1,275 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { AdminGetListResult } from "@gitpod/gitpod-protocol";
+import { useEffect, useRef, useState } from "react";
+import { getGitpodService } from "../service/service";
+import { PageWithAdminSubMenu } from "./PageWithAdminSubMenu";
+import { BlockedRepository } from "@gitpod/gitpod-protocol/lib/blocked-repositories-protocol";
+import ConfirmationModal from "../components/ConfirmationModal";
+import Modal from "../components/Modal";
+import CheckBox from "../components/CheckBox";
+
+export function BlockedRepositories() {
+    return (
+        <PageWithAdminSubMenu title="Blocked Repositories" subtitle="Search and manage all blocked repositories.">
+            <BlockedRepositoriesList />
+        </PageWithAdminSubMenu>
+    );
+}
+
+type NewBlockedRepository = Pick<BlockedRepository, "urlRegexp" | "blockUser">;
+
+interface Props {}
+
+export function BlockedRepositoriesList(props: Props) {
+    const [searchResult, setSearchResult] = useState<AdminGetListResult<BlockedRepository>>({ rows: [], total: 0 });
+    const [queryTerm, setQueryTerm] = useState("");
+    const [searching, setSearching] = useState(false);
+
+    const [isAddModalVisible, setAddModalVisible] = useState(false);
+    const [isDeleteModalVisible, setDeleteModalVisible] = useState(false);
+
+    const [currentBlockedRepository, setCurrentBlockedRepository] = useState({
```

Comment:
Can you do useState<BlockedRepository> and get rid of the need to cast the argument?

Reply:
Yes, make sense. Introduced the type `ExistingBlockedRepository` to make that work. :+1: